client/image_build.go

```diff
@@ -94,6 +94,7 @@ func (cli *Client) imageBuildOptionsToQuery(options types.ImageBuildOptions) (ur
 	query.Set("cgroupparent", options.CgroupParent)
 	query.Set("shmsize", strconv.FormatInt(options.ShmSize, 10))
 	query.Set("dockerfile", options.Dockerfile)
+	query.Set("ignorefile", options.Ignorefile)
```

Comment:
yes. I was thinking that `docker build --ignore-file .dockerignore` would do a couple of things:
1 - check for .dockerignore so that we can fail fast if its not there.  Note, it would only do this check if the user specified the --ignore-file flag.
2 - send the appropriate ignore-file query param to the daemon - but only when the user specified a value. Meaning, if they didn't specify the --ignore-file flag then there would be no query param.
3 - the daemon would detect whether or not the query param was specified. If specified, regardless of its value, if the file isn't there then it would generate an error.  If not there then we use the default value but do not generate an error if the file is missing.  This is similar to how we act on the client - no error for a missing file when we're defaulting, but generate an error when the user was explicit.